### PR TITLE
syncpoint (ticdc): set downstream tidb_enable_external_ts_read = off 

### LINKED ISSUE #7854: If tidb_enable_external_ts_read is set to on, Ticdc has no permission to synchronize data
### What did you do?

set global tidb_enable_external_ts_read=on;

### What did you expect to see?

Succeeded in synchronizing data.

### What did you see instead?

"code": "CDC:ErrMySQLTxnError",
        "message": "[CDC:ErrMySQLTxnError]MySQL txn error: Error 1105: only support read-only statement during read-only staleness transactions"

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):
V6.4.0
```console
(paste TiCDC version here)
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #7854 

#### Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Fix a bug that causes TiCDC can not replicate data downstream.
```

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

File: cdc/sink/mysql/mysql_syncpoint_store.go [hunk 1]
```diff
@@ -209,6 +210,19 @@ func (s *mysqlSyncPointStore) SinkSyncPoint(ctx context.Context,
 		log.Error("sync table: begin Tx fail", zap.Error(err))
 		return cerror.WrapError(cerror.ErrMySQLTxnError, err)
 	}
+
+	// we set `tidb_enable_external_ts_read` = off for each txn,
+	// otherwise TiCDC may not be able to write data to downstream TiDB when
+	// the downstream tidb_enable_external_ts_read = on.
+	if err = s.setTidbEnableExternalTsReadOff(ctx, tx); err != nil {
```
Comment (sdojjy) [inline cdc/sink/mysql/mysql_syncpoint_store.go:217]: maybe impact the dc mysql sink performance

Comment (nongfushanquan) [reply cdc/sink/mysql/mysql_syncpoint_store.go:217]: it will affect the performance of mysql sink more than sync_store

**Review by sdojjy — APPROVED** [review]

**Pull request merged**